src/main/scala/io/renku/jsonld/ontology/Ontology.scala

```diff
@@ -200,16 +201,16 @@ object ObjectProperty {
 
   implicit lazy val encoder: JsonLDEncoder[ObjectProperty] = JsonLDEncoder.instance {
     case ObjectProperty(id, range, domain, maybeTopProperty, maybeComment) =>
-      JsonLD.entity(
-        id,
-        EntityTypes of owl / "ObjectProperty",
-        Seq(
-          Some(rdfs / "domain" -> domain.asJsonLD),
-          Some(rdfs / "range"  -> range.asJsonLD),
-          maybeTopProperty.map(p => rdfs / "subPropertyOf" -> p.asJsonLD),
-          maybeComment.map(c => rdfs / "comment" -> c.asJsonLD)
-        ).flatten.toMap
-      )
+        JsonLD.entity(
+          id,
+          EntityTypes of owl / "ObjectProperty",
+          Seq(
+            Some(rdfs / "domain" -> domain.asJsonLD),
+            Some(rdfs / "range"  -> range.asJsonLD),
+            maybeTopProperty.map(p => rdfs / "subPropertyOf" -> p.encoder(p)),
```

Comment:
Just a question: why is here a `p.encoder(p)` necessary and where does `.encoder` come from? The line below is left as `.asJsonLD` so was wondering about this line.

Reply:
If I leave the original line (with `p.asJsonLD`) I step on a scala 2.12 bug and receive this error:
`could not find implicit value for parameter encoder: io.renku.jsonld.JsonLDEncoder[p.type]`

That's because scala 2.12 can't resolve `implicitly[p.type =:= TopObjectProperty]` (can only summon `implicitly[p.type <:< TopObjectProperty]`) so it seems `p.type` is constructed by the compiler as being a subtype of `TopObjectProperty` instead of it's identical type, hence the summon `p.asJsonLD` isn't working.

However, the alternative `p.encoder(p)` is identical, because the `encoder` implicit function lies in `TopObjectProperty` (which is the underlying type of `p`), so all I did was just use the summon by hand

Reply:
Thanks for the good explanation. 👍🏼 Strange thing that the compiler can't figure it out, I guess it is a bit difficult that `TopObjectProperty` is a type alias to a object type, not sure :) It's fine to me to leave it like this then.

Reply:
Yes, strange, and I think it's about a bug scala 2.12 had around singletons iirc, didn't manage to find it but didn't search too long anyway... It was something like this: a subtype of a singleton type should have the same type as his supertype (because a singleton can only have one inhabitant).

I think it applies here because the type `TopObjectProperty`, which is defined as `TopObjectProperty.type` is a singleton, and our culprit `p` inhabits the same type.

It was solved long ago but apparently after 2.12